collection-scripts/gather_network_trace

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+BASE_COLLECTION_PATH="must-gather"
+NETWORK_TRACE_PATH=${OUT:-"${BASE_COLLECTION_PATH}/network_trace"}
+
+mkdir -p ${NETWORK_TRACE_PATH}/
+
+# Convert the date and add the prefix
+function convert_log_timestamp {
+  grep -e "^I" -e "^E" -e "^W" -e "^N" | \
+  sed 's/ /-/;s/^[IEWDN]/& /' | \
+  gawk '{ print $2  " | "  $0; }' | \
+  sed "s/|/ ${1} ${2} ${3} | /" | \
+  sed '/^[+a-zA-Z/]/d'
+}
+
+function convert_ovs_timestamp {
+  sed 's/2020-//;s/-//;s/T/-/;s/Z//' | \
+  sed "s/|/ ${1} ${2} ${3} | /" | \
+  sed '/^[+a-zA-Z/]/d'
+}
+
+# Collect the log files
+function gather_ovn_kubernetes_data {
```

Comment:
Within the gather_networking script we have a block where this could be run, why add an extra script when this could be added to that file?

Reply:
There is post processing that converts the date to a common format and adds the node, pod, and container as a prefix to the line.

Reply:
Ultimately all logs are catted together and sorted bu the prefix to get the trace.

Reply:
A second point is we need to know which node a pod is running on, the ovs/ovn/ovn-master are tightly coupled. Some connections are unix sockets so we need to know which container is talking to which container on a node. When raft selects a new master we need to make sure the corresponding containers are running properly.

Reply:
With the basic trace we can add lines to the various logs that help is tracking progress of operations. For example noting when an item is created, used, and destroyed. We can find races using this and we can find out how long things take especially when several containers are involved. For example when an IP is assigned to a pod, it goes through 2 paths, one is kubelet creates the pod and crio/cni grabs the IP from ovn and the second is getting the IP into the ovn SDB and available to crio/cni. If crio asks before its available we get CrashLoopBackoff. The trace lets us track when the key events happen and how long things take to get done.

Reply:
I don't think Bim questioning the value, what I am asking is if this can be condensed into the other scripts were already supporting.

Reply:
@sferich888 Do you mind changing each log line with the prefix and common time stamp? Part of this is relating the container to the node, which you usually don't do. Also, when we modify existing files there is more testing to do. This can be done at this stage if you think it is important. Which scripts are you thinking about? I will take a look.

Reply:
I think I'm game. But I am concerned about the implications and performance impacts of such processing.

As you say, this will require additional testing. 
Let's explore this after we prove out the impacts with OVN. 

Can we rename this file to be explicitly for ovn_tracing?

Reply:
OK Changed the name to gather_network_ovn_trace